src/NServiceBus.Core.Tests/Scheduler/ScheduledTaskMessageHandlerTests.cs

```diff
@@ -36,7 +37,8 @@ public void When_a_scheduledTask_message_is_handled_the_task_should_be_defer()
                 TaskId = taskId
             }, handlingContext);
 
-            Assert.That(((ScheduledTask)handlingContext.DeferedMessage).TaskId, Is.EqualTo(taskId));
+            var deferredMessage = handlingContext.SentMessages.First(message => message.Options.GetDeliveryDelay().HasValue).Message;
```

Comment:
I think you can use `var deferredMessage = handlingContext.SentMessages.First(message => message.Options.GetDeliveryDelay().HasValue).Message<ScheduledTask>();` to avoid casting on the next line. Or the `Containing` extension method.